bench_test.go

```diff
@@ -0,0 +1,287 @@
+package honu_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
```

Comment:
Sure, I'm curious, could you explain what these global variables are are doing?

Reply:
The primary compiler optimization we're concerned with here is function inlining. When a function is called in Go it must assign a return address, create a stack location and put local variables on it, and specify the registers being modified - this can be a lot of work. The compiler optimizes this by replacing the function with the function's content inline, preventing a function call. This is great for production code, but for benchmarking it means that we're not getting a true view of how much time each operation would take when it is called. 

There are several rules for when a function is inlined, and for larger functions it's unlikely that the function will be inlined; however, it's easier to prevent the function from being inlined when benchmarking as a rule. The global variable means that the function has to access memory in the heap so it can't use the stack for inlining.

Reply:
oh, That makes sense, thanks for explaining!